src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -70,6 +70,14 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes()
                     response = new HttpResponseMessage(statusCode);
                     response.Content = new StringContent(body);
 
+                    // we default the Content-Type here, but we override below with any
+                    // Content-Type header the user might have set themselves
+                    if (Utility.IsJson(body))
```

Comment:
Yes, we actually have a very old work item that captures this work: https://github.com/Azure/azure-webjobs-sdk-script/issues/20

I'll also add a TODO in the code to revisit this